spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/repository/query/SpannerStatementQueryExecutor.java

```diff
@@ -303,18 +308,23 @@ private static void bindParameter(ValueBinder<Statement.Builder> bind,
 
 	public static String getColumnsStringForSelect(
 			SpannerPersistentEntity<?> spannerPersistentEntity, SpannerMappingContext mappingContext) {
+		return String.join(", ", spannerPersistentEntity.columns())
+				+ getChildrenSubquery(spannerPersistentEntity, mappingContext);
+	}
+
+	private static String getChildrenSubquery(
+			SpannerPersistentEntity<?> spannerPersistentEntity, SpannerMappingContext mappingContext) {
 		StringJoiner joiner = new StringJoiner(", ");
 		spannerPersistentEntity.doWithInterleavedProperties(spannerPersistentProperty -> {
 			if (spannerPersistentProperty.isEagerInterleaved()) {
 				Class childType = spannerPersistentProperty.getColumnInnerType();
 				SpannerPersistentEntity childPersistentEntity = mappingContext.getPersistentEntity(childType);
 				joiner.add(getChildrenStructsQuery(
-					childPersistentEntity, spannerPersistentEntity, mappingContext, spannerPersistentProperty.getColumnName()));
+						childPersistentEntity, spannerPersistentEntity, mappingContext, spannerPersistentProperty.getColumnName()));
 			}
 		});
-		String childrenSubquery = joiner.toString();
-		return String.join(", ", spannerPersistentEntity.columns())
```

Comment:
why was `String.join(", ", spannerPersistentEntity.columns())` removed?

Reply:
It is not removed - I've split an old method `getColumnsStringForSelect` on two - a new method `getChildrenSubquery` + old method `getColumnsStringForSelect`
So the old method now looks like
```
	public static String getColumnsStringForSelect(
			SpannerPersistentEntity<?> spannerPersistentEntity, SpannerMappingContext mappingContext) {
		return String.join(", ", spannerPersistentEntity.columns())
				+ getChildrenSubquery(spannerPersistentEntity, mappingContext);
	}
```

and all "joins" are in place

Reply:
I see, makes sense